dlib/test/trust_region.cpp

```diff
@@ -58,7 +58,7 @@ namespace
     {
         print_spinner();
 
-        matrix<T,2,1> ans;
+        matrix<T,0,1> ans;
```

Comment:
Now test fail, but if I do either
```c++
matrix<T,0,1> ans(2, 1);
```
or
```c++
matrix<T,0,1> ans;
ans.set_size(2, 1);
```
Then it doesn't build for me again... :/

Reply:
Hmm, ok. I wish I could reproduce [the original warning](https://github.com/davisking/dlib/pull/2490#discussion_r800173298) locally :-/

I believe I am doing exactly [this](https://github.com/davisking/dlib/pull/2490#discussion_r800176376) (=build from `dlib/tests`, using gcc 11.1.0).

Reply:
@arrufat this is just a shot in the dark, but how about: `matrix<T,0,1> ans = { 1,1 };`

Reply:
same :(

Reply:
Hmm, now I get a somewhat similar looking error:

```
/mnt/test/dlib/examples/svm_c_ex.cpp: In function ‘int main()’:
/mnt/test/dlib/examples/svm_c_ex.cpp:83:36: error: ‘normalizer.dlib::vector_normalizer<dlib::matrix<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout> >::temp_out.dlib::matrix<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>::data.dlib::row_major_layout::layout<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, 1>::data[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
   83 |     vector_normalizer<sample_type> normalizer;
      |                                    ^~~~~~~~~~
/mnt/test/dlib/examples/svm_c_ex.cpp:83:36: error: ‘*(double*)((char*)&normalizer + offsetof(dlib::vector_normalizer<dlib::matrix<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout> >,dlib::vector_normalizer<dlib::matrix<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout> >::temp_out.dlib::matrix<double, 2, 1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>::<unnamed>))’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
make[2]: *** [examples/examples_build/CMakeFiles/svm_c_ex.dir/build.make:63: examples/examples_build/CMakeFiles/svm_c_ex.dir/svm_c_ex.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:790: examples/examples_build/CMakeFiles/svm_c_ex.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```